src/sentry/api/endpoints/accept_project_transfer.py

```diff
@@ -105,12 +103,18 @@ def post(self, request: Request) -> Response:
             return Response({"detail": "Invalid organization"}, status=400)
 
         # check if user is an owner of the organization
-        is_org_owner = OrganizationMember.objects.filter(
+        query = OrganizationMember.objects.filter(
             user__is_active=True,
             user=request.user,
-            role=roles.get_top_dog().id,
             organization_id=organization.id,
-        ).exists()
+        )
+        is_org_owner = (
+            query.filter(role=roles.get_top_dog().id).exists()
+            or OrganizationMemberTeam.objects.filter(
+                team__in=organization.get_teams_with_org_role(roles.get_top_dog().id),
+                organizationmember_id__in=list(query.values_list("id", flat=True)),
+            ).exists()
```

Comment:
I think there should be an abstracted method _somewhere_ to check if a user/orgmember has a role, either from individual assignment or group assignment. 

People are unlikely going to remember to filter for both so it is better to have a method to do it across the codebase.

Reply:
> People are unlikely going to remember to filter for both so it is better to have a method to do it across the codebase.


:100: Having team roles available in commonly used abstractions like `request.access` and simple to operate methods to check permissions will be important as the additional logic for team roles are not simple.